sdks/router-sdk/src/entities/mixedRoute/route.ts

```diff
@@ -54,17 +55,22 @@ export class MixedRouteSDK<TInput extends Currency, TOutput extends Currency> {
       const prevPool = pools[i - 1]
       const pool = pools[i]
       const inputToken = tokenPath[i]
+      const isWrappedNativeEquality = ((pool.token0.equals(inputToken.wrapped) || pool.token1.equals(inputToken.wrapped)) && inputToken.isNative)
+      || ((pool.token0.wrapped.equals(inputToken) && pool.token0.isNative) || (pool.token1.wrapped.equals(inputToken) && pool.token1.isNative))
 
       const outputToken =
         pool instanceof V4Pool
-          ? pool.token0.equals(inputToken)
+          ? (isWrappedNativeEquality
+            || pool.token0.equals(inputToken))
             ? pool.token1
             : pool.token0
           : pool.token0.wrapped.equals(inputToken.wrapped)
           ? pool.token1
           : pool.token0
 
-      invariant(isValidTokenPath(prevPool, pool, inputToken), `PATH`)
+      if (!isWrappedNativeEquality) {
```

Comment:
yeah this makes sense to me we want to add an invariant in the isWrappedNativeEquality case?

Reply:
i feel like probs not necessary actually, not really sure what we would check in any case